### ISSUE TITLE
Odd results when using Lunr v2.3.0 with lunr-languages

### ISSUE THREAD

**Issue by yeraydiazdiaz** [olivernn/lunr.js#358]
Hi Oliver,

while porting Lunr.py to include the recent changes in Lunr v2.3.0 I noticed a big difference between the results when using lunr-languages on the same corpus from v2.2.1 and below.

I put together a couple of codesandboxes to showcase the issue, on both I'm using the same corpus of spanish documents and searching for the same term "imperio":
- [Using v2.3.0](https://codesandbox.io/embed/9l618l40po?hidenavigation=1)
- [Using v2.2.1](https://codesandbox.io/embed/n0y1qxozn4?hidenavigation=1)

Note both the order and the scores are different. The difference in the scores is to be expected, but the order of the documents what's interesting. Here's a summary of the term presence in the corpus:

- `e` contains the term in the title and 2 times in the body
- `d` contains the term 3 times in the body
- `f` and `g` contain the term once in the body

v2.2.1 returns `e`, `d`, `f`, `g`.
v2.3.0 returns `d`, `f`, `g`, `e`.

Note `e` scores last on v2.3.0 which is unexpected.

The results in English between the the Javascript and Python versions still match closely in v2.3.0, it's only when using language support that they diverge. Lunr.py targetting v2.3.0 with language support returns results closer to v2.2.1:

```
{'ref': 'd', 'score': 1.047, 'match_data': <MatchData "imperi">},
{'ref': 'e', 'score': 0.738, 'match_data': <MatchData "imperi">},
{'ref': 'f', 'score': 0.732, 'match_data': <MatchData "imperi">},
{'ref': 'g', 'score': 0.679, 'match_data': <MatchData "imperi">}
```

Though I'm not sure why `e` scores so low.

/cc @MihaiValentin

**Comment by yeraydiazdiaz** [comment]
Glad you liked the test data 😄  I figured it'd be easier to understand the text if it was something familiar.

You're right, turns out I expected the stemmer to turn `imperial` and `imperiales`, the two instances in `e`, to `imperi` which is the stem of `imperio`, turns out it doesn't, hence `e` scoring lower.

After tracing both versions it appears the changes in `Vector.similarity` make up for all the difference, given that the results in v2.3.0 are actually more coherent I'm going to close this.

Thanks for your help 🙂

**Comment by olivernn** [comment]
Yeah, I think the way the similarity method worked before was _probably_ wrong. I mean, it worked, but made boosts, either at query-time or build-time, mostly ineffective.

I'm glad the new scoring/ranking seems to be more coherent, let me know if anything else crops up while your upgrading Lunr.py.

**Comment by yeraydiazdiaz** [comment]
That was the only “issue” actually, everything else went smoothly. Just released [0.4.0](https://pypi.org/project/lunr/#history) targeting Lunr.js 2.3.0 🙂
